ui/utils/logout.ts

```diff
@@ -8,29 +8,37 @@ export async function handleLogout() {
 
   const oidc = await oidcSource();
 
-  if (oidc.isEnabled()) {
-    const session = await getSession();
-    const idToken = session?.idToken;
-
-    const discovery = await fetch(`${oidc.provider!.wellKnown}`);
-    const discoveryDoc = await discovery.json();
-
-    const endSessionEndpoint = discoveryDoc.end_session_endpoint;
-    if (endSessionEndpoint && idToken) {
-      const logoutUrl = `${endSessionEndpoint}?id_token_hint=${encodeURIComponent(
-        idToken,
-      )}&post_logout_redirect_uri=${encodeURIComponent(
-        window.location.origin + "/api/auth/oidc/signin",
-      )}`;
-
-      // Clear local session without redirect flicker
-      await signOut({ redirect: false });
-
-      // Then navigate browser to Keycloak logout page
-      window.location.href = logoutUrl;
-      return;
+  try {
+    // Check if OIDC is configured
+    if (oidc.isEnabled()) {
+      const session = await getSession();
+      const idToken = session?.idToken;
+
+      const discovery = await fetch(`${oidc.provider!.wellKnown}`);
+      const discoveryDoc = await discovery.json();
+
+      const endSessionEndpoint = discoveryDoc.end_session_endpoint;
+
+      if (endSessionEndpoint) {
+        // Build logout URL (with or without id_token_hint)
+        const logoutUrl = `${endSessionEndpoint}${
+          idToken ? `?id_token_hint=${encodeURIComponent(idToken)}&` : "?"
```

Comment:
What if the `idToken` is null/undefined here? Will it just be an empty parameter?